antarest/study/business/link_management.py

```diff
@@ -52,38 +56,73 @@ class LinkManager:
     def __init__(self, storage_service: StudyStorageService) -> None:
         self.storage_service = storage_service
 
-    def get_all_links(self, study: RawStudy, with_ui: bool = False) -> t.List[LinkInfoDTO]:
+    def get_all_links(self, study: RawStudy, with_ui: bool = False) -> t.List[LinkInfoDTOType]:
         file_study = self.storage_service.get_storage(study).get_raw(study)
-        result = []
+        result: t.List[LinkInfoDTOType] = []
+
         for area_id, area in file_study.config.areas.items():
-            links_config: t.Optional[t.Dict[str, t.Any]] = None
-            if with_ui:
-                links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+            links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+
             for link in area.links:
-                ui_info: t.Optional[LinkUIDTO] = None
-                if with_ui and links_config and link in links_config:
-                    ui_info = LinkUIDTO(
-                        color=f"{links_config[link].get('colorr', '163')},{links_config[link].get('colorg', '163')},{links_config[link].get('colorb', '163')}",
-                        width=links_config[link].get("link-width", 1),
-                        style=links_config[link].get("link-style", "plain"),
+                link_properties = links_config[link]
+
+                link_creation_data: t.Dict[str, t.Any] = {"area1": area_id, "area2": link}
+                link_creation_data.update(link_properties)
+                if not with_ui:
+                    link_creation_data.update(
+                        {
+                            "colorr": None,
+                            "colorb": None,
+                            "colorg": None,
+                            "link-width": None,
+                            "link-style": None,
+                        }
                     )
-                result.append(LinkInfoDTO(area1=area_id, area2=link, ui=ui_info))
+
+                link_data: LinkInfoDTOType
+                if int(study.version) < 820:
+                    link_data = LinkInfoDTOBase(**link_creation_data)
+                else:
+                    link_data = LinkInfoDTO820(**link_creation_data)
+
+                result.append(link_data)
 
         return result
 
-    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTO) -> LinkInfoDTO:
+    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTOType) -> LinkInfoDTOType:
+        if link_creation_info.area1 == link_creation_info.area2:
+            raise LinkValidationError("Cannot create link on same node")
```

Comment:
IMO it's better to log `area` instead of `node` and also add the concerned area name